bin/pyspark

```diff
@@ -42,11 +42,10 @@ if [[ -z "$PYSPARK_DRIVER_PYTHON" ]]; then
   PYSPARK_DRIVER_PYTHON="${PYSPARK_PYTHON:-"python"}"
 fi
 
-WORKS_WITH_IPYTHON=$(python -c 'import sys; print(sys.version_info >= (2, 7, 0))')
-
 # Determine the Python executable to use for the executors:
 if [[ -z "$PYSPARK_PYTHON" ]]; then
-  if [[ $PYSPARK_DRIVER_PYTHON == *ipython* && ! $WORKS_WITH_IPYTHON ]]; then
+  WORKS_WITH_IPYTHON=$(python -c 'import sys; print(sys.version_info >= (2, 7, 0))')
```

Comment:
Let's just remove this check. Honestly, I don't think it makes much sense to check executor's Python version by driver's `python`.

Reply:
cc @JoshRosen

Reply:
Agree, this doesn't really need an explicit check, not just for ipython.

Reply:
@srowen   you mean that there should no version checks whatsoever, or the checks should not be about IPython, just about python >= 2.7.

Something like for example
```bash
if [[ -z "$PYSPARK_PYTHON" ]]; then
  PYSPARK_PYTHON=python
fi
if [[ $($PYSPARK_PYTHON -c 'import sys; print(sys.version_info >= (2, 7, 0))') != "True" ]]; then
  echo "PYSPARK_PYTHON ($PYSPARK_PYTHON) should be Python 2.7 or higher" 1>&2
  exit 1
fi

if [[ -z "$PYSPARK_DRIVER_PYTHON" ]]; then
  PYSPARK_DRIVER_PYTHON=$PYSPARK_PYTHON
fi
if [[ $($PYSPARK_DRIVER_PYTHON -c 'import sys; print(sys.version_info >= (2, 7, 0))') != "True" ]]; then
  echo "PYSPARK_DRIVER_PYTHON ($PYSPARK_DRIVER_PYTHON) should be Python 2.7 or higher" 1>&2
  exit 1
fi
```

Reply:
I think this is just obsolete, as Spark has required 2.7 for a while. More goes wrong if you don't use 2.7, so, not much point in maintaining this extra check.